mmdata.c

```diff
@@ -1121,26 +1104,24 @@ long nmbrRevInstr(long start_position, nmbrString *string1,
     nmbrString *string2)
 {
    long ls1, ls2;
-   nmbrString *tmp = NULL_NMBRSTRING;
    ls1 = nmbrLen(string1);
    ls2 = nmbrLen(string2);
    if (start_position > ls1 - ls2 + 1) start_position = ls1 - ls2 + 2;
    if (start_position<1) return 0;
    while (!nmbrEq(string2, nmbrMid(string1, start_position, ls2))) {
      start_position--;
-     nmbrLet(&tmp, NULL_NMBRSTRING);
-              /* Clear nmbrString buffer to prevent overflow caused by "mid" */
+     nmbrTempAlloc(0); /* Clear temporaries to prevent overflow caused by "mid" */
```

Comment:
Dont know whether this is correct.

Reply:
This pattern of clearing an already cleared variable was the previous idiom for calling `let` for its side effect of clearing the temp buffer. I think that it is better not to involve a dead variable and confuse the reader. In this case it is pretty easy to see that `tmp` is otherwise useless: it is only ever set to `NULL_NMBRSTRING`.

Reply:
Again: this frees temporaries allocated by callers of this function, too.  How is it guaranteed this never occurs now and in future?

Reply:
Note that `nmbrTempAlloc(0)` appears literally at the end of `nmbrLet()`. This does not reset the stack to 0, it resets the stack to `g_nmbrStartTempAllocStack`, so this will actually work even if `g_nmbrStartTempAllocStack` is nonzero, although as I mention in another comment this only happens in exceptional situations, and most functions have the precondition that `g_nmbrStartTempAllocStack` is zero.